draft-ietf-quic-recovery.md

```diff
@@ -302,32 +302,31 @@ Retransmission Timeout mechanisms.
 ### Crypto Handshake Timeout
 
 Data in CRYPTO frames is critical to QUIC transport and crypto negotiation, so a
-more aggressive timeout is used to retransmit it.  Below, the term "handshake
-packet" is used to refer to packets containing CRYPTO frames, not packets with
-the specific long header packet type Handshake.
+more aggressive timeout is used to retransmit it.  Below, the term "crypto
+packet" is used to refer to packets containing CRYPTO data sent in Initial or
+Handshake packets.
 
-The initial handshake timeout SHOULD be set to twice the initial RTT.
+The initial crypto retransmission timeout SHOULD be set to twice the initial
+RTT.
 
 At the beginning, there are no prior RTT samples within a connection.  Resumed
 connections over the same network SHOULD use the previous connection's final
-smoothed RTT value as the resumed connection's initial RTT.
-
-If no previous RTT is available, or if the network changes, the initial RTT
-SHOULD be set to 100ms.
-
-When CRYPTO frames are sent, the sender SHOULD set a timer for the handshake
-timeout period.  Upon timeout, the sender MUST retransmit all unacknowledged
-CRYPTO data by calling RetransmitAllUnackedHandshakeData(). On each
-consecutive expiration of the handshake timer without receiving an
-acknowledgement for a new packet, the sender SHOULD double the handshake timeout
-and set a timer for this period.
+smoothed RTT value as the resumed connection's initial RTT.  If no previous RTT
+is available, or if the network changes, the initial RTT SHOULD be set to 100ms.
+When an acknowledgement is received, a new RTT is computed and the timer
+SHOULD be set for twice the newly computed smoothed RTT.
+
+When CRYPTO frames are sent in Initial or Handshake packets, the sender SHOULD
+set a timer for the crypto timeout period.  Upon timeout, the sender MUST
+retransmit all unacknowledged CRYPTO data by calling
+RetransmitAllUnackedCryptoData(). On each consecutive expiration of the
+crypto timer without receiving an acknowledgement for a new packet, the
+sender SHOULD double the crypto handshake timeout and set a timer for this
+period.
 
 When CRYPTO frames are outstanding, the TLP and RTO timers are not active unless
 the CRYPTO frames were sent at 1-RTT encryption.
```

Comment:
Now that you have a definition for "crypto packets" above which excludes 1-RTT, you could probably simplify this by saying that "When crypto packets are outstanding, the TLS and RTO timers are not active."

Reply:
TLS->TLP